codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -886,7 +895,27 @@ let checkContainsRegisteredContractAddress = (self: t, ~contractName, ~contractA
 /**
 * Returns the latest block number fetched for the lowest fetcher queue (ie the earliest un-fetched dynamic contract)
 */
-let getLatestFullyFetchedBlock = (self: t) => self.baseRegister.latestFetchedBlock
+let getLatestFullyFetchedBlock = (self: t) => {
+  //Consider pending dynamic contracts when calculating the latest fully fetched block
+  //Since they are now registered lazily on query or update of the fetchstate, not when
+  //the register function is called
+  let minPendingDynamicContracts = self.pendingDynamicContracts->Belt.Array.reduce(None, (
+    acc,
+    contract,
+  ) => {
+    let {registeringEventBlockNumber} = contract
+    minOfOption(registeringEventBlockNumber - 1, acc)->Some
+  })
+
+  switch (self.baseRegister.latestFetchedBlock, minPendingDynamicContracts) {
+  | ({blockNumber}, Some(pendingDynamicContractBlockNumber))
+    if pendingDynamicContractBlockNumber < blockNumber => {
+      blockNumber: pendingDynamicContractBlockNumber,
+      blockTimestamp: 0,
```

Comment:
Is it fine to have 0 here?

Reply:
Yes, this is the same behaviour as before where when you create a dynamic contract register it initialises with blockTimestamp 0 since we don't know what the timestamp is for that block.